src/update-process-env.js

```diff
@@ -121,13 +121,22 @@ async function getEnvFromShell(env) {
   }
 
   let result = {};
+  let skip = false;
   for (let line of stdout.split('\n')) {
-    if (line.includes('=')) {
+    // start of shell function definition: skip full definition
+    if (line.includes('=() {')) {
```

Comment:
What if a variable value contains the character sequence "=() {" ?
i.e. `x='a=() { b }'`

Reply:
// end of shell function definition
(line==='}'){
skip=false;